### PR TITLE
feat: add ability to filter by group tag

### LINKED ISSUE #878: feat: support group tags when filtering `checkly test --tags`
### What problem does this feature solve?

Currently the `--tags` flag for `npx checkly test` only applies to check tags. We should make the filtering also apply to group tags. If a group matches the tags filter, then all checks in the group should be run as well.

This is particularly important for users using glob patterns to create browser checks in a group, since there's no way to set tags on the automatically generated browser checks.

### PR DESCRIPTION
**Pull request by ferrandiaz**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [ ] Create CLI
* [x] Test
* [ ] Docs
* [ ] Examples
* [ ] Other

## Notes for the Reviewer

The check only provides groupId information, hence the part of trying to get the group from the check ref

> Resolves #878

### CODE REVIEW

File: packages/cli/src/commands/test.ts [hunk 1]
```diff
@@ -181,7 +181,13 @@ export default class Test extends AuthCommand {
         return filterByCheckNamePattern(grep, check.name)
       })
       .filter(([, check]) => {
-        return filterByTags(targetTags?.map((tags: string) => tags.split(',')) ?? [], check.tags)
+        const tags = check.tags ??= []
```
Comment (umutuzgur) [inline packages/cli/src/commands/test.ts:184]: Why did you go with `??=`? This assigns an empty array to check.tags if it is empty first and then assigns the same value to the `tags` variable. I'm curious if there was a special reason

Comment (umutuzgur) [reply packages/cli/src/commands/test.ts:184]: Then I would suggest just using `??` to not modify the check and groups data, even though it wouldn't be a big problem 🤷‍♂️

Comment (ferrandiaz) [reply packages/cli/src/commands/test.ts]: yup, did that, thanks for the tip!

**Review by umutuzgur — APPROVED** [review]
LGTM

**Pull request merged**